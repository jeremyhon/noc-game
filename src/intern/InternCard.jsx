import React, { PropTypes } from 'react'
import { Card, CardTitle, CardText } from 'material-ui/Card'
import map from 'lodash/map'
import skillPropType from '../lib/skill'

const renderGenderIcon = (gender) => {
  if (gender === 'M') {
    return <i className="fa fa-mars" aria-hidden="true" />
  }
  return <i className="fa fa-venus" aria-hidden="true" />
}

const renderTitle = (name, gender) => (<span>{name} {renderGenderIcon(gender)}</span>)
const renderSkills = (skills) => map(skills, (skill, key) => (<li key={key}>{skill}</li>))

const InternCard = (props) => (
  <Card className="intern-card">
    <CardTitle title={renderTitle(props.name, props.gender)} />
    <CardText>
      <ul>
        {renderSkills(props.skills)}
      </ul>
    </CardText>
  </Card>
)

InternCard.propTypes = {
  name: PropTypes.string.isRequired,
  gender: PropTypes.oneOf(['M', 'F']).isRequired,
  skills: PropTypes.arrayOf(skillPropType).isRequired,
}

export default InternCard
