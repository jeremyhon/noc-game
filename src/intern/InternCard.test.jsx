import React from 'react'
import { shallow } from 'enzyme'

import InternCard from './InternCard'

it("matches the snapshot", () => {
  expect(shallow(<InternCard />)).toMatchSnapshot()
})
