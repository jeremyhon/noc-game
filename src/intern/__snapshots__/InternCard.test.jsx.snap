// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Card
    className="intern-card"
    skills={undefined}
    subtitle="Skills"
    title={
        <span>
             
            <i
                aria-hidden="true"
                className="fa fa-venus"
            />
        </span>
    }
/>,
  "nodes": Array [
    <Card
      className="intern-card"
      skills={undefined}
      subtitle="Skills"
      title={
            <span>
                   
                  <i
                        aria-hidden="true"
                        className="fa fa-venus"
                  />
            </span>
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <InternCard />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Card
          className="intern-card"
          skills={undefined}
          subtitle="Skills"
          title={
                    <span>
                               
                              <i
                                        aria-hidden="true"
                                        className="fa fa-venus"
                              />
                    </span>
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <Card
          className="intern-card"
          skills={undefined}
          subtitle="Skills"
          title={
                    <span>
                               
                              <i
                                        aria-hidden="true"
                                        className="fa fa-venus"
                              />
                    </span>
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InternCard />,
}
`;
